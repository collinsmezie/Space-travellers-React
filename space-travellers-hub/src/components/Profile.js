import React, {useEffect} from 'react';
import { useSelector, useDispatch } from 'react-redux';
import ListGroup from 'react-bootstrap/ListGroup';
import { fetchMissionData } from '../redux/Missions/missions';
import { fetchRocketData } from '../redux/Rockets/rockets';

const Profile = () => {
    const { missions} = useSelector((state) => state.missionReducer);
    const { rockets } = useSelector((state) => state.rocketsReducer);
    const dispatch = useDispatch();

    useEffect(() => {
        if (missions.length === 0)
            dispatch(fetchMissionData());

        if (rockets.length === 0)
            dispatch(fetchRocketData());
    }, []);

    const filteredMissions = missions.filter(mission => mission.mission_status === true);
    const filteredRockets = rockets.filter(rocket => rocket.reserved === true);
    return (
        <div className="flexy">
        {missions.length === 0 && <p>There are no Missions to display!</p>}
            <div>
                <h3>My Missions</h3>
                <ListGroup>
                    {
                        filteredMissions.length === 0 && (<p>Nobody has joined any mission!</p>)
                    }
                    {
                        filteredMissions.map(filteredMission => {
                            return <ListGroup.Item key={filteredMission.mission_id} className="filter-width" >{filteredMission.mission_name}</ListGroup.Item>
                        })
                    }
                </ListGroup>
            </div>

        {rockets.length === 0 && <p>There are no Rockets to display!</p>}
        <div>
            <h3>My Rockets</h3>
            <ListGroup>
                {
                    filteredRockets.length === 0 && (<p>No Rocket reservations yet!</p>)
                }
                {
                    filteredRockets.map(filteredRocket => {
                        return <ListGroup.Item key={filteredRocket.id}>{filteredRocket.name}</ListGroup.Item>
                    })
                }
            </ListGroup>
        </div>
        </div>
    );
}

export default Profile