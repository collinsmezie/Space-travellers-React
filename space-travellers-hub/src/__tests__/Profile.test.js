import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Profile from '../components/Profile';
import store from '../redux/configureStore';

describe('Test: Profile Component', () => {
  test('should render', () => {
    const profile = renderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(profile).toMatchSnapshot();
  });
});
