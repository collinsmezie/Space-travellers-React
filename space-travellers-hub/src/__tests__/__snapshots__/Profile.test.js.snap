// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Profile Component should render 1`] = `
Array [
  <div
    className="flexy"
  >
    <p>
      There are no Missions to display!
    </p>
    <div>
      <h3>
        My Missions
      </h3>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <p>
          Nobody has joined any mission!
        </p>
      </div>
    </div>
    <p>
      There are no Rockets to display!
    </p>
    <div>
      <h3>
        My Rockets
      </h3>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <p>
          No Rocket reservations yet!
        </p>
      </div>
    </div>
  </div>,
  ",",
]
`;
